src/metabase/pulse/util.clj

```diff
@@ -45,24 +46,49 @@
       (catch Throwable e
         (log/warnf e "Error running query for Card %s" card-id)))))
 
-(defn execute-multi-card
-  "Multi series card is composed of multiple cards, all of which need to be executed.
+(defn is-card-empty?
+  "Check if the card is empty"
+  [card]
+  (if-let [result (:result card)]
+    (or (zero? (-> result :row_count))
+        ;; Many aggregations result in [[nil]] if there are no rows to aggregate after filters
+        (= [[nil]]
+           (-> result :data :rows)))
+    ;; Text cards have no result; treat as empty
+    true))
 
-  This is as opposed to combo cards and cards with visualizations with multiple series,
-  which are viz settings."
-  [card-or-id dashcard-or-id]
-  (if dashcard-or-id
-    (let [card-id     (u/the-id card-or-id)
-          card        (t2/select-one :model/Card :id card-id, :archived false)
-          ;; NOTE/TODO - dashcard-or-id is nil with multiple time series
-          dashcard-id (u/the-id dashcard-or-id)
-          dashcard    (t2/select-one :model/DashboardCard :id dashcard-id)
-          multi-cards (dashboard-card/dashcard->multi-cards dashcard)]
-      (for [multi-card (if (seq multi-cards)
-                         multi-cards
-                         [card])]
-        (execute-card {:creator_id (:creator_id card)} (:id multi-card))))
-    (let [card-id (u/the-id card-or-id)
-          ;; NOTE/TODO - dashcard-or-id is nil with multiple time series
-          card    (t2/select-one :model/Card :id card-id, :archived false)]
-      [(execute-card {:creator_id (:creator_id card)} (:id card))])))
+(defn execute-dashboard-subscription-card
+  "Returns subscription result for a card.
+
+  This function should be executed under pulse's creator permissions."
+  [dashcard parameters]
+  (let [{card-id      :card_id
+         dashboard-id :dashboard_id} dashcard]
+    (try
+      (let [card           (t2/select-one :model/Card :id card-id)
+            multi-cards    (dashboard-card/dashcard->multi-cards dashcard)
+            result-fn      (fn [card-id]
+                             {:card     (if (= card-id (:id card))
+                                          card
+                                          (t2/select-one :model/Card :id card-id))
+                              :dashcard dashcard
+                              :type     :card
```

Comment:
I'd not die on this hill, but what about introducing a new type of `multi-cards` so we could differentiate between a single card and a card that has multi-series results?

I don't like the fact that we're assuming a card and a multi-card is the samething.

Reply:
Yea, I've been thinking about that too. I'd like to merge this PR first, though.

I'm also working on multi-series cards csv/xlsx downloads/subscription attachments, where it will probably be much better to have mulit-cards as their own thing, like you suggest. 

So, for this PR, I think no, but in a next step, yea, almost certainly a good idea.